runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java

```diff
@@ -84,9 +86,22 @@
   private MetricsMap<KV<MetricName, HistogramData.BucketType>, HistogramCell> histograms =
       new MetricsMap<>(HistogramCell::new);
 
-  /** Create a new {@link MetricsContainerImpl} associated with the given {@code stepName}. */
+  /**
+   * Create a new {@link MetricsContainerImpl} associated with the given {@code stepName}. If
+   * stepName is null, this MetricsContainer is not bound to a step.
+   */
   public MetricsContainerImpl(@Nullable String stepName) {
     this.stepName = stepName;
+    this.isProcessWide = false;
```

Comment:
this(stepName, false)

Reply:
(no change done)
Do you want me to add another private ctor?
Currently there is no ctor defined to call "this(stepName, false)"

Reply:
I must have been looking at a previous iteration of this code (or getting it mixed up with others).